tests/composition-active-with-melt.cc

```diff
@@ -0,0 +1,178 @@
+#include <aspect/material_model/interface.h>
+#include <aspect/melt.h>
+#include <deal.II/base/parameter_handler.h>
+
+/**
+ * This material model extends the simple material model
+ * to include melt transport.
+ */
+
+namespace aspect
+{
+  using namespace dealii;
+
+  template <int dim>
+  class SimpleWithMelt:
+    public MaterialModel::Interface<dim>
+  {
+    public:
+      virtual bool
+      viscosity_depends_on (const MaterialModel::NonlinearDependence::Dependence dependence) const
```

Comment:
All these depends_on functions are deprecated or removed long ago. Lets try to get rid of them, despite copy-pasting from old models